src/main/java/Application/OpenCVServer/OpenCVServerOperator.java

```diff
@@ -0,0 +1,141 @@
+        in = new DataInputStream(clientSocket.getInputStream());
+    }
+
+    // change name to something more appropriate
+    public void standbyForConnection() throws IllegalArgumentException, IOException {
+        //if input is faulty, timeout and continue
+        currentImage = ImageIO.read(ImageIO.createImageInputStream(clientSocket.getInputStream()));
+        ImageIO.write(currentImage, "png", new File(fileName));
+        Mat subject = ImageProcessor.getImageMat(fileName);
+
+        detector.detect(subject);
+        String ids = OpenCVUtil.serializeMat(detector.getIds());
+        String corners = OpenCVUtil.serializeMatList(detector.getCorners());
+        String result = "";
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode resultObject = mapper.createObjectNode();
+        resultObject.put("ids", ids);
+        resultObject.put("corners", corners);
+        System.out.println("responding");
+
+        result = mapper.writeValueAsString(resultObject);
+
+        out.println(result);
+
+    }
+
+    public void stopConnection() throws IOException {
+        in.close();
+        out.close();
+        clientSocket.close();
+        serverSocket.close();
+    }
+
+    public static OpenCVServerOperator getInstance() {
+        if (instance == null) {
+            instance = new OpenCVServerOperator();
+        }
+        return instance;
+    }
+
+    public void setupServerRunners(String[] args) throws IllegalArgumentException {
+        String[] argument;
+        String masterUrl = ""; // for test: http://host.docker.internal:4567/
+        String beatCommand = ""; // for test: client/register
+        String serverId = "";
+        String serverIp = "";
+        boolean connected = false;
+
+        serverRunner = new ServerRunner();
+
+        List<String> missingParameterList = new ArrayList<>(List.of("SERVER_ID", "SERVER_IP", "MASTER_URL", "BEAT_COMMAND", "PORT")); // Connected has default value of false
+
+        for (String arg : args) {
+            argument = arg.split("=");
+            if (argument.length == 2) {
+                switch (argument[0]) {
+                    case "SERVER_ID":
+                        serverId = argument[1];
+                        missingParameterList.remove("SERVER_ID");
+                        break;
+                    case "SERVER_IP":
+                        serverIp = argument[1];
+                        missingParameterList.remove("SERVER_IP");
+                        break;
+                    case "MASTER_URL":
+                        masterUrl = argument[1];
+                        missingParameterList.remove("MASTER_URL");
+                        break;
+                    case "BEAT_COMMAND":
+                        beatCommand = argument[1];
+                        missingParameterList.remove("BEAT_COMMAND");
+                        break;
+                    case "CONNECTED":
```

Comment:
Why is that a parameter?

Reply:
I put it as a parameter because beat command and get server command dependes on how it is implemented in the Master, so I wanted to give an access to change it without changing the code
I'm considering to add a TODO to move it to a config file

Reply:
yes, just add a TODO to move it to a config file